File: Current/Product/Production/Nant/Common.Tasks/SaveProperty.vb
Option Explicit On
Option Strict On

Imports NAnt.Core
Imports NAnt.Core.Attributes

Public Class SaveProperty
    Inherits Element

    Private _PropertyName As String
    Private _PropertyValue As String

    Public ReadOnly Property PropertyValue() As String
        Get
            If Not Me._PropertyValue Is Nothing Then
                Return Me._PropertyValue
            End If
            Return Me.Properties(Me.PropertyName)
        End Get
    End Property

    <TaskAttribute("name", Required:=True)> _
    Public Property PropertyName() As String
        Get
            Return _PropertyName
        End Get
        Set(ByVal value As String)
            _PropertyName = value
        End Set
    End Property

    <TaskAttribute("value", Required:=False)> _
    Public WriteOnly Property SetProperty() As String
        Set(ByVal value As String)
            Me._PropertyValue = value
        End Set
    End Property

End Class

File: tags/0.9.0.95/Product/Production/Nant/CIFactory.NAnt.Tasks/Types/SaveProperty.cs
using System;
using System.Collections;
using System.Collections.Generic;
using NAnt.Core;
using NAnt.Core.Attributes;

namespace CIFactory.NAnt.Types
{
    public class SaveProperty : Element
    {
        #region Fields

        private string _PropertyName;

        private string _PropertyValue;

        #endregion

        #region Properties

        [TaskAttribute("name", Required = true)]
        public string PropertyName
        {
            get { return _PropertyName; }
            set { _PropertyName = value; }
        }

        public string PropertyValue
        {
            get
            {
                if (this._PropertyValue != null)
                {
                    return this._PropertyValue;
                }
                return this.Properties[this.PropertyName];
            }
        }

        [TaskAttribute("value", Required = false)]
        public string SetProperty
        {
            set { this._PropertyValue = value; }
        }

        #endregion

    }
}